var/spack/repos/builtin/packages/turbovnc/package.py

```diff
@@ -0,0 +1,95 @@
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the terms and
+# conditions of the GNU Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public
+# License along with this program; if not, write to the Free Software
+# Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA 02111-1307 USA
+##############################################################################
+from spack import *
+
+
+class Turbovnc(Package):
+    """TurboVNC is a derivative of VNC (Virtual Network Computing) 
+ that is tuned to provide peak performance for 3D and video workloads.
+ TurboVNC was originally a fork of TightVNC 1.3.x, on the surface, 
+ the X server and Windows viewer still behave similarly to their parents."""
+    # FIXME: add a proper url for your package's homepage here.
+    homepage = "http://www.turbovnc.org/"
+    url      = "http://downloads.sourceforge.net/project/turbovnc/2.0.1/turbovnc-2.0.1.tar.gz"
+
+
+    version('2.0.1', 'a279fdb9ac86a1ebe82f85ab68353dcc')
+    version('2.0.91', '1e203cc533103cf7bd1186c7fd620186',
+            url      = "http://downloads.sourceforge.net/project/turbovnc/2.0.91%20%282.1beta2%29/turbovnc-2.0.91.tar.gz"            
+    )
+    version('2.1', '6748bb13647d318f0c932394f8298d10')
+
+    # FIXME: Add dependencies if this package requires them.
+    variant('java', default=False, description='Enable Java build')
+
+   
+    depends_on('cmake', type='build')
+    depends_on("libjpeg-turbo")
+    depends_on("openssl")
+    depends_on("pam")
+    depends_on("libx11")
+    depends_on("libxext")
+
+    def url_for_version(self, version):
+        """Handle TurboVNC's version-based custom URLs."""
+        return 'http://downloads.sourceforge.net/project/turbovnc/%s/turbovnc-%s.tar.gz' % (
+            version, version)
+
+    def validate(self, spec):
+        """
+        Checks if incompatible versions of openssl were specified
+
+        :param spec: spec of the package
+        :raises RuntimeError: in case of inconsistencies
+        """
+
+        if spec.satisfies('@:2.1') and spec.satisfies('^openssl@1.1:'):
+            msg = 'turbovnc does not compile with openssl 1.1 '
+            raise RuntimeError(msg)
+
+    def install(self, spec, prefix):
+        
+        self.validate(spec)
+
+        def feature_to_bool(feature, on='ON', off='OFF'):
+            if feature in spec:
+                return on
+            return off
+#        layout = YamlDirectoryLayout(self.tmpdir)
+#        rel_path=layout.relative_path_for_spec(spec)
+        feature_args = []
+        # FIXME: Modify the configure line to suit your build system here.
+        if '+java' not in spec:
+            feature_args.append(
+                '-DTVNC_BUILDJAVA=%s' % feature_to_bool('+java'))
+            feature_args.append(
+                '-DTVNC_BUILDNATIVE=%s' % feature_to_bool('+java'))
+        feature_args.append('-DCMAKE_VERBOSE_MAKEFILE=ON')
```

Comment:
This is a general CMake feature, Spack will control it as needed (no need to put in packages).